test/e2e/flowaggregator_test.go

```diff
@@ -238,59 +262,76 @@ func checkRecordsForFlows(t *testing.T, data *TestData, srcIP string, dstIP stri
+				checkFlowType(t, record, ipfixregistry.FlowTypeIntraNode)
+			} else {
+				checkPodAndNodeData(t, record, "perftest-a", controlPlaneNodeName(), "perftest-c", workerNodeName(1))
+				checkFlowType(t, record, ipfixregistry.FlowTypeInterNode)
+			}
+
+			if checkService {
 				if isIntraNode {
-					checkPodAndNodeData(t, record, "perftest-a", controlPlaneNodeName(), "perftest-b", controlPlaneNodeName())
-					checkFlowType(t, record, ipfixregistry.FlowTypeIntraNode)
+					if !strings.Contains(record, "antrea-test/perftest-b") {
+						t.Errorf("Record with ServiceIP does not have Service name")
+					}
 				} else {
-					checkPodAndNodeData(t, record, "perftest-a", controlPlaneNodeName(), "perftest-c", workerNodeName(1))
-					checkFlowType(t, record, ipfixregistry.FlowTypeInterNode)
-				}
-
-				if checkService {
-					if isIntraNode {
-						if !strings.Contains(record, "antrea-test/perftest-b") {
-							t.Errorf("Record with ServiceIP does not have Service name")
-						}
-					} else {
-						if !strings.Contains(record, "antrea-test/perftest-c") {
-							t.Errorf("Record with ServiceIP does not have Service name")
-						}
+					if !strings.Contains(record, "antrea-test/perftest-c") {
+						t.Errorf("Record with ServiceIP does not have Service name")
 					}
 				}
-				if checkNetworkPolicy {
-					// Check if records have both ingress and egress network policies.
-					if !strings.Contains(record, ingressNetworkPolicyName) {
-						t.Errorf("Record does not have NetworkPolicy name with ingress rule")
-					}
-					if !strings.Contains(record, fmt.Sprintf("%s: %s", "ingressNetworkPolicyNamespace", testNamespace)) {
-						t.Errorf("Record does not have correct ingressNetworkPolicyNamespace")
-					}
-					if !strings.Contains(record, egressNetworkPolicyName) {
-						t.Errorf("Record does not have NetworkPolicy name with egress rule")
-					}
-					if !strings.Contains(record, fmt.Sprintf("%s: %s", "egressNetworkPolicyNamespace", testNamespace)) {
-						t.Errorf("Record does not have correct egressNetworkPolicyNamespace")
-					}
+			}
+			if checkNetworkPolicy {
+				// Check if records have both ingress and egress network policies.
+				if !strings.Contains(record, ingressNetworkPolicyName) {
+					t.Errorf("Record does not have NetworkPolicy name with ingress rule")
+				}
+				if !strings.Contains(record, fmt.Sprintf("%s: %s", "ingressNetworkPolicyNamespace", testNamespace)) {
+					t.Errorf("Record does not have correct ingressNetworkPolicyNamespace")
+				}
+				if !strings.Contains(record, egressNetworkPolicyName) {
+					t.Errorf("Record does not have NetworkPolicy name with egress rule")
+				}
+				if !strings.Contains(record, fmt.Sprintf("%s: %s", "egressNetworkPolicyNamespace", testNamespace)) {
+					t.Errorf("Record does not have correct egressNetworkPolicyNamespace")
+				}
+			}
+			if checkBandwidth && !strings.Contains(record, "octetDeltaCount: 0") {
```

Comment:
Is not 0 delta count for ipfix control flow? If so, please add a comment here.

Reply:
Correct, thanks for reminding.